tests/integration_test.go

```diff
@@ -0,0 +1,90 @@
+package tests
+
+import (
+	"bytes"
+	"io/ioutil"
+	"path"
+	"path/filepath"
+	"testing"
+
+	skynet "github.com/NebulousLabs/go-skynet"
+	"gopkg.in/h2non/gock.v1"
+)
+
+// TestUploadAndDownloadFile tests uploading and downloading a single file.
+func TestUploadAndDownloadFile(t *testing.T) {
+	defer gock.Off() // Flush pending mocks after test execution
+
+	const srcFile = "../testdata/file1.txt"
+	const skylink = skynet.URISkynetPrefix + "testskynet"
+
+	// Test upload.
+
+	gock.New(skynet.DefaultUploadOptions.PortalURL).
+		Post(skynet.DefaultUploadOptions.PortalUploadPath).
+		Reply(200).
+		JSON(map[string]string{"skylink": "testskynet"})
+
+	skylink2, err := skynet.UploadFile(srcFile, skynet.DefaultUploadOptions)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if skylink2 != skylink {
```

Comment:
Does this test pass? From reading the code I would expect it to fail because the mocked response is `{"skylink": "testskynet"}` and not `{"skylink": "sia://testskynet"}` which is what you are asserting, right?

Reply:
It passes because `UploadFile` prepends the sia prefix to the response.

Reply:
Oh ok, well then this is a great test actually ^^